src/popup/useApplier.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+// when apply function is called
+// applyRequired will be true next render
+// it's used for syncing state with storage/localStorage only in certain moments
+export const useApplier = () => {
+  const applyRequired = useRef(false);
+
+  const apply = useCallback(() => {
+    applyRequired.current = true;
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
```

Comment:
Is it really intended to be in setTimeout without time?

Reply:
Yes, but it works even without it ...